implementations/micrometer-registry-new-relic/src/main/java/io/micrometer/newrelic/NewRelicInsightsAgentClientProvider.java

```diff
@@ -181,7 +181,7 @@ public void publish(NewRelicMeterRegistry meterRegistry) {
     @Override
     public Map<String, Object> writeFunctionTimer(FunctionTimer timer) {
         Map<String, Object> attributes = new HashMap<>();
-        TimeUnit timeUnit = TimeUnit.valueOf(timer.getId().getBaseUnit());
+        TimeUnit timeUnit = TimeUnit.valueOf(timer.getId().getBaseUnit().toUpperCase());
```

Comment:
For `Timer` and `FunctionTimer`, there is a `baseTimeUnit` method that will return the base TimeUnit directly without having to do this. `LongTaskTimer` does not have such a method, which may have just been an oversight.

Reply:
I've updated those two.  Would the addition of LongTaskTimer getBaseTimeUnit(), AbstractLongTaskTimer and simple updates to newLongTaskTimer() methods in some meter registries be too much for this?  I have a branch with all tests working.

Reply:
@shakuzen  See PR https://github.com/micrometer-metrics/micrometer/pull/1944 for adding baseTimeUnit to LongTaskTimer.

Reply:
Thank you for making it as a separate pull request. That would have been my preference anyways.